### PR TITLE
Adding reserve limits to master nodes.

### LINKED ISSUE #357: add limits on master nodes
see #258. @mjudeikis @kwoodson what's your opinion?

### PR DESCRIPTION
**Pull request by kwoodson**
Adding system-reserve limits to master nodes.

fixes #357

### CODE REVIEW

**Comment by kwoodson** [comment]
After running the sync manually the configmaps were updated and the master's node-config contains the reserved values.
```
[root@master-000000 ~]# grep -A 1 reserved /etc/origin/node/node-config.yaml
  system-reserved:
  - cpu=1000m,memory=1Gi
```

File: pkg/addons/data/ConfigMap/openshift-node/node-config-master.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ data:
       - /etc/origin/node/pods
       rotate-certificates:
       - "true"
+      system-reserved:
+      - cpu=1000m,memory=1Gi
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/addons/data/ConfigMap/openshift-node/node-config-master.yaml]: we already do this via https://github.com/openshift/openshift-azure/blob/master/pkg/addons/data/ConfigMap/openshift-node/node-config-compute.yaml#L40 so derivative function need to be added to this one too.

**Comment by kwoodson** [comment]
[root@master-000000 ~]# grep -A 1 reserved /etc/origin/node/node-config.yaml 
  system-reserved:
  - cpu=500m,memory=1Gi

File: pkg/api/default.go [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
+		AgentPoolProfileRoleMaster: {
+			"system-reserved": "cpu=500m,memory=1Gi",
+		},
+		AgentPoolProfileRoleCompute: {
+			"kube-reserved":   "cpu=200m,memory=512Mi",
+			"system-reserved": "cpu=200m,memory=512Mi",
+		},
+		AgentPoolProfileRoleInfra: {
+			"kube-reserved":   "cpu=200m,memory=512Mi",
+			"system-reserved": "cpu=200m,memory=512Mi",
```
Comment (mjudeikis) [inline pkg/api/default.go]: as discussed this is a little bit of duplication, but I think we soon will deviate from same values on infra nodes and compute (as soon as our nodes will start running 100 pods and infra still on 10). So this leaves us a place to change. 
I already see a follow-up story. 

We need to monitor system-reserved, kube-reserved and overall node memory blueprint in our stacks to see if these values make sense.

**Comment by kwoodson** [comment]
@jim-minter @pweil- @kargakis , could use a review if available, thanks.

**Comment by kwoodson** [comment]
Resolved merge conflicts.

File: pkg/addons/data/ConfigMap/openshift-node/node-config-master.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjudeikis) [reply pkg/addons/data/ConfigMap/openshift-node/node-config-master.yaml]: replace this with `*** GENERATE ***`

[the remark this answers is not in the document]
Comment (kwoodson) [reply pkg/addons/data/ConfigMap/openshift-node/node-config-master.yaml]: This is quite frustrating.  vscode editor shows this change in my workspace but on file it is not updated.  I have forced it and repushed.

**Comment by mjudeikis** [comment]
`make generate` 
@jim-minter @kargakis @pweil- ptal. I think this is good to merge now.

**Comment by kwoodson** [comment]
regenerated.

**Comment by kwoodson** [comment]
squashed commits.

File: pkg/api/types.go [hunk 3]
```diff
@@ -120,6 +120,16 @@ const (
 	OSTypeWindows OSType = "Windows"
 )
 
+// ReservedResources represents the type of reserved resource
+type ReservedResources string
```
Comment (pweil-) [inline pkg/api/types.go]: nit: make singular

changes LGTM

Comment (kwoodson) [reply pkg/api/types.go]: Updated.

**Comment by kwoodson** [comment]
Will squash commits.

**Comment by kwoodson** [comment]
@pweil-, updated and squashed.  Does this look better?

**Pull request merged**